lib/commands/recordscreen.js

```diff
@@ -0,0 +1,161 @@
+import _ from 'lodash';
+import { sleep } from 'asyncbox';
+import helpers from '../android-helpers';
+
+let commands = {}, extensions = {};
+
+/**
+* record the display of devices running Android 4.4 (API level 19) and higher.
+* It records screen activity to an MPEG-4 file. Audio is not recorded with the video file.
+*
+* @param filePath the video file name
+*                 for example, "/sdcard/demo.mp4"
+* @param videoSize the format is widthxheight.
+*                  if it is "default", the default value is the device's native display resolution (if supported),
+*                  1280x720 if not. For best results,
+*                  use a size supported by your device's Advanced Video Coding (AVC) encoder.
+*                  for example, "1280x720"
+* @param timeLimit the maximum recording time, in seconds. if it is -1, the default and maximum value is 180 (3 minutes).
+* @param bitRate the video bit rate for the video, in megabits per second.
+*                if it is -1, the default value is 4Mbps. You can increase the bit rate to improve video quality,
+*                but doing so results in larger movie files.
+*                for example, 6000000
+*
+*/
+commands.startRecordingScreen = async function ( filePath, videoSize, timeLimit, bitRate ) {
+  let cmd, data, length, indexNoSuchFile;
+  let loop = true;
+  let opts = {};
+  let {udid, emPort} = await helpers.getDeviceInfoFromCaps(opts);
+  
+  opts.udid = udid;
+  opts.emPort = emPort;
+
+  let adb = await helpers.createADB(opts.javaVersion,
+                                    opts.udid,
+                                    opts.emPort,
+                                    opts.adbPort,
+                                    opts.suppressKillServer);
+
+  data = await adb.getApiLevel();
+
+  //this function is suppported on the device running android 4.4(api level 19)
+  if (data < 19) throw new Error('The screenrecord command is able to record the display of devices running Android 4.4 (API level 19) and higher.');
+
+  cmd = ['ls', `${filePath}`, '|', 'grep', 'No such file'];
+  data = await adb.shell(cmd); 
+
+  length = _.size(data);
+
+  //if there's same file in the path, then thorws error
+  if (length === 0) throw new Error(`${filePath} is already exist.`);
+
+
+  //make adb command 
+  cmd = ['screenrecord', `${filePath}`];
+
+  if ( videoSize !== "default" ) {
+    cmd.push('--size', videoSize);
+  }  
+
+  if ( timeLimit !== -1 ) {
+    cmd.push('--time-limit', timeLimit);
+  } 
+
+  if ( bitRate !== -1 ) {
+    cmd.push('--bit-rate', bitRate);
+  } 
+
+  //call the command  
+  data = adb.shell(cmd); 
+
+  
+  //there is the delay time to start recording the screen, so, wait until it is ready.
+  //the ready condition is
+  //1. check the movie file is created
+  //2. check it is started to capture the screen 
+  cmd = ['ls', '-al', `${filePath}`]; 
+  while (loop) {
```

Comment:
This infinite loop will not work for us. If for any reason there is an error, this will just keep going and going.

Parsing the output like this is not going to fly. Try something like this:
```js
  await retryInterval(10, 1000, async () => {
    let data = await adb.shell(cmd);
    if (data.indexOf('No such file') !== -1) {
      throw new Error('No video file found');
    }
    // https://regex101.com/r/mwmAhQ/1
    let match = /^[-rwx]{10}\s+\S+\s+\S+\s+(\d+)\s+/.exec(data.trim());
    if (!match) {
      throw new Error(`Unable to get list data for file '${filePath}'`);
    }
    let size = parseInt(match[1], 10);
    if (size < 32) {
      throw new Error(`File '${filePath}' is not yet full created`);
    }
  });
```